From 03bdd7f2fe4491eaed0432336dcaae4a020a753f Mon Sep 17 00:00:00 2001
From: Dong Aisheng <b29396@freescale.com>
Date: Thu, 25 Sep 2014 15:14:16 +0800
Subject: [PATCH 0506/1074] ENGR00333130-2 dts: imx6sx: add legacy imx6sx sdb
 revA support

The CAN transceiver is changed on RevB board and the default imx6sx-sdb.dts
is for support new RevB board.
This patch adds the dts for legacy RevA board support, especially for CAN
device.
This is for people who still wants to use RevA board with this code base.

Signed-off-by: Dong Aisheng <b29396@freescale.com>
(cherry picked from commit 9d90c66e3cf61ed243ae148d401f2b46875f0745)
[Original patch taken from git://git.freescale.com/imx/linux-2.6-imx.git]
Signed-off-by: Biyao Zhai <biyao.zhai@windriver.com>
---
 arch/arm/boot/dts/Makefile                  |    2 +
 arch/arm/boot/dts/imx6sx-sdb-canfd-reva.dts |   35 +++++++++++++++++++++++++++
 arch/arm/boot/dts/imx6sx-sdb-reva.dts       |   24 ++++++++++++++++++
 arch/arm/boot/dts/imx6sx-sdb.dts            |    8 +++---
 4 files changed, 65 insertions(+), 4 deletions(-)
 create mode 100644 arch/arm/boot/dts/imx6sx-sdb-canfd-reva.dts
 create mode 100644 arch/arm/boot/dts/imx6sx-sdb-reva.dts

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index 953132e..85b1897 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -201,11 +201,13 @@ dtb-$(CONFIG_ARCH_MXC) += \
 	imx6sx-19x19-arm2-sai.dtb \
 	imx6sx-sdb.dtb \
 	imx6sx-sdb-ldo.dtb \
+	imx6sx-sdb-reva.dtb \
 	imx6sx-sdb-lcdif1.dtb \
 	imx6sx-sdb-sai.dtb \
 	imx6sx-sdb-emmc.dtb \
 	imx6sx-sdb-m4.dtb \
 	imx6sx-sdb-canfd.dtb \
+	imx6sx-sdb-canfd-reva.dtb \
 	imx6sx-sabreauto.dtb \
 	vf610-cosmic.dtb \
 	vf610-twr.dtb
diff --git a/arch/arm/boot/dts/imx6sx-sdb-canfd-reva.dts b/arch/arm/boot/dts/imx6sx-sdb-canfd-reva.dts
new file mode 100644
index 0000000..8e8766d
--- /dev/null
+++ b/arch/arm/boot/dts/imx6sx-sdb-canfd-reva.dts
@@ -0,0 +1,35 @@
+/*
+ * Copyright (C) 2014 Freescale Semiconductor, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "imx6sx-sdb-reva.dts"
+
+/*
+ * flexcan and canfd are sharing the same pads, so disable flexcan to
+ * enable canfd.
+ * There's a trick here that the tranceiver configuration gpios are still
+ * got from the flexcan node since they're using the same tranceiver.
+ */
+&flexcan1 {
+	status = "disabled";
+};
+
+&flexcan2 {
+	status = "disabled";
+};
+
+&canfd1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_canfd1_1>;
+	status = "okay";
+};
+
+&canfd2 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_canfd2_1>;
+	status = "okay";
+};
diff --git a/arch/arm/boot/dts/imx6sx-sdb-reva.dts b/arch/arm/boot/dts/imx6sx-sdb-reva.dts
new file mode 100644
index 0000000..0089232
--- /dev/null
+++ b/arch/arm/boot/dts/imx6sx-sdb-reva.dts
@@ -0,0 +1,24 @@
+/*
+ * Copyright (C) 2014 Freescale Semiconductor, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+
+#include "imx6sx-sdb.dts"
+
+/*
+ * The CAN transceiver is changed on RevB board.
+ * This dts is used for support the old CAN transceiver on RevA board
+ */
+
+&flexcan1 {
+	trx-en-gpio = <&gpio4 25 GPIO_ACTIVE_HIGH>;
+	trx-stby-gpio = <&gpio4 27 GPIO_ACTIVE_HIGH>;
+};
+
+&flexcan2 {
+	trx-en-gpio = <&gpio4 25 GPIO_ACTIVE_LOW>;
+	trx-stby-gpio = <&gpio4 27 GPIO_ACTIVE_LOW>;
+};
diff --git a/arch/arm/boot/dts/imx6sx-sdb.dts b/arch/arm/boot/dts/imx6sx-sdb.dts
index 60fffbb..2bdb01b 100644
--- a/arch/arm/boot/dts/imx6sx-sdb.dts
+++ b/arch/arm/boot/dts/imx6sx-sdb.dts
@@ -311,16 +311,16 @@
 &flexcan1 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_flexcan1_1>;
-	trx-en-gpio = <&gpio4 25 GPIO_ACTIVE_HIGH>;
-	trx-stby-gpio = <&gpio4 27 GPIO_ACTIVE_HIGH>;
+	trx-en-gpio = <&gpio4 25 GPIO_ACTIVE_LOW>;
+	trx-stby-gpio = <&gpio4 27 GPIO_ACTIVE_LOW>;
 	status = "okay";
 };
 
 &flexcan2 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_flexcan2_1>;
-	trx-en-gpio = <&gpio4 25 GPIO_ACTIVE_HIGH>;
-	trx-stby-gpio = <&gpio4 27 GPIO_ACTIVE_HIGH>;
+	trx-en-gpio = <&gpio4 25 GPIO_ACTIVE_LOW>;
+	trx-stby-gpio = <&gpio4 27 GPIO_ACTIVE_LOW>;
 	status = "okay";
 };
 
-- 
1.7.5.4

